app/knight.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
+        self.knight_property = knight
+        self.name = knight["name"]
+        self.power = 0
+        self.hp = 0
+        self.protection = 0
+
+    def is_item(self, key: str) -> bool:
+        if self.knight_property[key] is not None:
+            return True
+        return False
+
+    def set_power(self) -> int:
+        return (self.knight_property["power"]
+                + (self.knight_property["weapon"].get("power", 0)
+                   if self.is_item("weapon") else 0)
+                + (self.knight_property["potion"]["effect"].get("power", 0)
+                   if self.is_item("potion") else 0))
+
+    def set_hp(self) -> int:

```

Comment:
Code Style Issue: Similar to the `set_power` method, the `set_hp` method does not actually set the hp attribute of the knight. Consider renaming it to `calculate_hp`.